xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -793,21 +798,141 @@ HRESULT CVideoBufferShared::GetResource(ID3D11Resource** ppResource)
   {
     // open resource on app device
     ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
-    hr = pD3DDevice->OpenSharedResource(handle, __uuidof(ID3D11Resource), &m_sharedRes);
+    if (FAILED(hr = pD3DDevice->OpenSharedResource(handle, __uuidof(ID3D11Resource), &m_sharedRes)))
+    {
+      CLog::LogF(LOGDEBUG, "unable to open the shared resource, error description: {}",
+                 DX::GetErrorDescription(hr));
+      return hr;
+    }
+  }
+
+  if (FAILED(hr = m_sharedRes.CopyTo(ppResource)))
+    return hr;
+
+  // Failures in the synchronization code are not blocking problems, log but return success
+  if (!m_appFence && m_handleFence != INVALID_HANDLE_VALUE)
+  {
+    // cache the context and open the fence on app device
+    ComPtr<ID3D11DeviceContext1> context1 = DX::DeviceResources::Get()->GetImmediateContext();
+    ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
+    ComPtr<ID3D11Device5> device5;
+    if (FAILED(hr = context1.As(&m_appContext4)))
+    {
+      CLog::LogF(LOGDEBUG, "ID3D11DeviceContext4 is not available, error description: {}",
+                 DX::GetErrorDescription(hr));
+    }
+    else if (FAILED(hr = pD3DDevice.As(&device5)))
+    {
+      CLog::LogF(LOGDEBUG, "ID3D11Device5 is not available, error description: {}",
+                 DX::GetErrorDescription(hr));
+      m_appContext4 = nullptr;
+    }
+    else if (FAILED(hr = device5->OpenSharedFence(m_handleFence, IID_PPV_ARGS(&m_appFence))))
+    {
+      CLog::LogF(LOGDEBUG, "unable to open the shared fence, error description: {}",
+                 DX::GetErrorDescription(hr));
+      m_appContext4 = nullptr;
+    }
   }
 
-  if (SUCCEEDED(hr))
-    hr = m_sharedRes.CopyTo(ppResource);
+  if (m_appFence)
+  {
+    // Make the GPU wait for the fence value that produced the picture
+    ComPtr<ID3D11DeviceContext1> context1 = DX::DeviceResources::Get()->GetImmediateContext();
+    ComPtr<ID3D11DeviceContext4> context4;
```

Comment:
`context1` and `context4` are unused variables now

Reply:
yes, multitasking is no good. I'll get there.